### DISCUSSION TITLE
Primitive types (int, float etc.) java code generation

### DISCUSSION THREAD

**Discussion by kspmmitt** [square/wire#3059] [Q&A]
Hello,
As seen while generating java classes for a proto3 schema, wire compiler not generating wrapper types for primitives like int, float etc. But if given the field in proto3 schema as "optional" then it generates primitive type as wrapper class in java.
This understanding is correct?
Also anyway to generate always as wrapper rather then primitive?
Rgds,
Kamal

**Comment by oldergod** [comment]
This is on purpose and the change is from https://github.com/square/wire/pull/1679

There's no way currently to force boxed types.
